app/ui/browser-modern/views/browser/tabbar/tab.jsx

```diff
@@ -34,8 +34,12 @@ const TAB_STYLE = Style.registerStyle({
   alignItems: 'center',
   overflow: 'hidden',
   boxSizing: 'border-box',
-  width: `${UIConstants.TAB_DEFAULT_WIDTH}px`,
-  minWidth: `${UIConstants.TAB_MIN_WIDTH}px`,
+  '&[data-mounted=true]': {
+    transition: 'width 0.2s ease-out',
+    width: `${UIConstants.TAB_DEFAULT_WIDTH}px`,
+    minWidth: `${UIConstants.TAB_MIN_WIDTH}px`,
```

Comment:
`minWidth` can be pulled out of the data-mounted selector and left next to the `width` property as it was before (since being mounted should have nothing to do with that property).  Also, `transition` could be added to the non data-mounted portion but up to you